MPP-2847 - Document the switch to optional TLS (#3316)
mozilla/fx-private-relay: docs/0001-optional-tls-for-incoming-smtp.md

> A publicly-referenced SMTP server MUST NOT require use of the
> STARTTLS extension in order to deliver mail locally. This rule
> prevents the STARTTLS extension from damaging the interoperability of
> the Internet's SMTP infrastructure. A publicly-referenced SMTP
> server is an SMTP server which runs on port 25 of an Internet host
> listed in the MX record (or A record if an MX record is not present)
> for the domain name on the right hand side of an Internet mail
> address.

Our inbound SMTP server, via `dig mozmail.com MX`, is provided by AWS SES:

```
mozmail.com.		600	IN	MX	10 inbound-smtp.us-west-2.amazonaws.com.
```

[EFF-STARTTLS]: https://www.eff.org/deeplinks/2014/11/starttls-downgrade-attacks
[GOOGLE-EMAIL]: https://transparencyreport.google.com/safer-email/overview
[RFC3207]: https://www.rfc-editor.org/rfc/rfc3207
[CENTRALOPS]: https://centralops.net/co/
[KICKBOX]: https://kickbox.com/

### Option 2: Switch to "Optional TLS" for incoming email

**Good** - Relay masks will be accepted on more websites. This will protect
our user's true email.

**Good** - Valid emails from servers and clients that do not implement TLS /
`STARTTLS` will now be forwarded to Relay users.

> **Bad** - Relay users will not know if an email was sent with TLS or not.
> Amazon SES does not provide this data. This prevents Relay from allowing users
> to decide about requiring TLS, or creating filter rules based on TLS.

Reviewer: quibble (non-blocking):
```suggestion
**Bad** - Relay users will not know if an email was sent with TLS or not.
Amazon SES does not provide this data. So Relay users can't require TLS
or create filter rules based on TLS.
```
